fastapi/utils.py

```diff
@@ -95,11 +106,15 @@ def create_response_field(
 def create_cloned_field(
     field: ModelField,
     *,
-    cloned_types: Optional[Dict[Type[BaseModel], Type[BaseModel]]] = None,
+    cloned_types: Optional[
+        MutableMapping[Type[BaseModel], Type[BaseModel]]
+    ] = WeakKeyDictionary(),
```

Comment:
This can easily be moved to a global variable for clarity if preferred.

Reply:
Thanks an absolute ton for the work on this issue/PR, it's super insightful and helpful!

What would your thoughts be on making the above move to a global variable, along with retaining the default `cloned_types = None`?

Benefits would include:
- one instance of `WeakKeyDictionary` instead of potentially multiple
    - eg, if `cloned_types` is just omitted, the default `WeakKeyDictionary()` gets instantiated 
    - if later anywhere, someone explicitly passes `cloned_types=None`, a new `WeakKeyDictionary()` gets instantiated each time this happens
- current tests would continue to pass
- current test coverage would remain at 100%
- not needing any further test investigation may help @tiangolo's ability to review + adopt this change

Eg, as far as I can tell, something like the following appears to fix the failing coverage issue in this way:
```
$ git diff
diff --git a/fastapi/utils.py b/fastapi/utils.py
index d9a7d2d4..f67cad5c 100644
--- a/fastapi/utils.py
+++ b/fastapi/utils.py
@@ -28,6 +28,9 @@ if TYPE_CHECKING:  # pragma: nocover
     from .routing import APIRoute
 
 
+CLONED_TYPES_CACHE = WeakKeyDictionary()
+
+
 def is_body_allowed_for_status_code(status_code: Union[int, str, None]) -> bool:
     if status_code is None:
         return True
@@ -111,12 +114,12 @@ def create_cloned_field(
     *,
     cloned_types: Optional[
         MutableMapping[Type[BaseModel], Type[BaseModel]]
-    ] = WeakKeyDictionary(),
+    ] = None,
 ) -> ModelField:
     # cloned_types caches already cloned types to support recursive models and improve
     # performance by avoiding unecessary cloning
     if cloned_types is None:
-        cloned_types = WeakKeyDictionary()
+        cloned_types = CLONED_TYPES_CACHE
 
     original_type = field.type_
     if is_dataclass(original_type) and hasattr(original_type, "__pydantic_model__"):
```

Reply:
Perhaps the green check mark will entice a review https://github.com/tiangolo/fastapi/pull/4645/commits/d1e71a05372d58c9fd6cda171dddf624210d3074